### PR TITLE
Update release notes - justification for changes in `v0.4.0` 

### PR DESCRIPTION
**Pull request by iameskild**
@dharhas @magsol @costrouc @tonyfast I have tried to outline some of the most compelling reasons why we decided to make the changes that we did. Let me know if you think we need to any or change anything.

### CODE REVIEW

**Comment by iameskild** [comment]
I went back and reformatted the dates for past release from `mm/dd/yyyy` to `month day, year` spelt out to avoid any confusion.

**Comment by danlester** [comment]
@iameskild Please note that you may want to rebase to main and then run the pre-commit hooks again because we have adjusted the config for mdformat slightly.

**Comment by costrouc** [comment]
Looks great to me. I have no problem with us merging this PR

**Comment by magsol** [comment]
Made some minor grammatical changes. I don't really have an answer for the email contact that we should provide. Otherwise, LGTM.

**Review by magsol — APPROVED** [review]

**Comment by dharhas** [comment]
We are not offering email support, they should be pointed to github issues/discussions

**Comment by viniciusdc** [comment]
We could point then to Gitter as well

File: RELEASE.md [hunk 1]
```diff
@@ -1,27 +1,52 @@
...
 
 ### Feature changes and enhancements
 
 ### Bug fixes
 
-## Release v0.4.0 - 03/17/2022
+## Release v0.4.0 - March 17, 2022
+
+**WARNING**
+> If you're looking for a stable version of QHub, please consider `v0.3.14`. The `v0.4.0` has many breaking changes and has several bugs that we are working to resolve in the upcoming releases.
```
Comment (dharhas) [inline RELEASE.md:16]: change to something like: `has rough edges that we will be smoothing over in upcoming point releases`

**Comment by dharhas** [comment]
We don't really want folks to use gitter. I'd rather get us into better habits in github

**Comment by dharhas** [comment]
This might help:

https://github.com/github/feedback/discussions/2844#discussioncomment-2061814

**Comment by magsol** [comment]
If we're moving away from gitter we may want to advertise that explicitly.
I'm all in favor of getting away from gitter but we should make that an
overt part of our strategy moving forward.

**Pull request merged**